### PR TITLE
Add lazy loading and suspense

### PR DESCRIPTION
**Pull request by pabloccid**
Added lazy loading in routes component.

### CODE REVIEW

File: generators/app/templates/src/app/components/Routes/index.ejs [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
 import { history } from '../../../redux/store';
-import Home from '../../screens/Dashboard';
+import Loading from '../Spinner/components/loading';
 
 import AuthenticatedRoute from './components/AuthenticatedRoute';
 import * as Routes from './constants';
 import styles from './styles.scss';
+
+const Home = lazy(() => import('../../screens/Dashboard'));
+
```
Comment (wfolini) [inline generators/app/templates/src/app/components/Routes/index.ejs]: Delete this space because the comment below add a extra space when is bootstrapped

File: generators/app/templates/src/app/components/Routes/index.ejs [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
 
 import { history } from '../../../redux/store';
-import Home from '../../screens/Dashboard';
+import Loading from '../Spinner/components/loading';
 
 import AuthenticatedRoute from './components/AuthenticatedRoute';
 import * as Routes from './constants';
 import styles from './styles.scss';
+
+const Home = lazy(() => import('../../screens/Dashboard'));
```
Comment (wfolini) [inline generators/app/templates/src/app/components/Routes/index.ejs:12]: Are we going to add a func like this for each screen that we add? I'd think of a different approach for this

**Review by wfolini — CHANGES_REQUESTED** [review]
Excellent contribution! ❤️ 👏 👏

File: generators/app/templates/src/app/components/Routes/index.ejs [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
+
 <%# TODO: Handle ConnectedRouter connection according redux feature %>
 function AppRoutes() {
   return (
     <ConnectedRouter history={history}>
       <div className={styles.container}>
-        <Switch>
-          <AuthenticatedRoute isPrivateRoute exact path={Routes.HOME} component={Home} />
-        </Switch>
+        <Suspense fallback={<Loading />}>
```
Comment (damfinkel) [inline generators/app/templates/src/app/components/Routes/index.ejs]: what about the minimum time to start showing the spinner?

File: generators/app/templates/src/app/components/Routes/index.ejs (hunk 2 above)
Comment (damfinkel) [reply generators/app/templates/src/app/components/Routes/index.ejs:12]: This is how is done. This replaces the import, so it won't have more lines thatn before, just instead of doing `import Home  from ...` you have to do `const Home = lazy(() => import(...))`

Comment (wfolini) [reply generators/app/templates/src/app/components/Routes/index.ejs:12]: Sure, you are right... I didn't notice that the imports aren't here :D

Comment (wfolini) [reply generators/app/templates/src/app/components/Routes/index.ejs:12]: Sure, you are right... I didn't notice that the normal imports aren't here :D

File: generators/app/templates/src/app/components/Suspense/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React, { Suspense } from 'react';
+import PropTypes from 'prop-types';
+
+import Loading from '../Spinner/components/loading';
+
+import { DEFAULT_MAX_DURATION } from './constants';
+
+function CustomSuspense({ maxDuration, fallback, children }) {
```
Comment (wfolini) [inline generators/app/templates/src/app/components/Suspense/index.js:8]: 👏

File: generators/app/templates/src/app/components/Routes/index.ejs [hunk 5]
```diff
@@ -1,22 +1,22 @@
-import React, { Suspense, lazy } from 'react';
+import React, { lazy } from 'react';
 import { ConnectedRouter } from 'connected-react-router';
 import { Switch } from 'react-router-dom';
 
 import { history } from '../../../redux/store';
-import Loading from '../Spinner/components/loading';
+import Suspense from '../Suspense';
+
```
Comment (wfolini) [inline generators/app/templates/src/app/components/Routes/index.ejs:7]: Double space?

**Review by wfolini — COMMENTED** [review]
👏

File: generators/app/templates/src/app/components/Suspense/index.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React, { Suspense } from 'react';
+import PropTypes from 'prop-types';
+
+import Loading from '../Spinner/components/loading';
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/Suspense/index.js:4]: `components/Loading` ?

File: generators/app/templates/src/app/components/Routes/index.ejs [hunk 7]
```diff
@@ -1,21 +1,25 @@
...
+import React, { lazy } from 'react';
 import { ConnectedRouter } from 'connected-react-router';
 import { Switch } from 'react-router-dom';
 
 import { history } from '../../../redux/store';
-import Home from '../../screens/Dashboard';
+import Suspense from '../Suspense';
 
 import AuthenticatedRoute from './components/AuthenticatedRoute';
 import * as Routes from './constants';
```
Comment (mvbattan) [inline generators/app/templates/src/app/components/Routes/index.ejs]: This will be needed to be changed in another PR, avoiding `import *`

**Review by mvbattan — APPROVED** [review]
Great !

File: generators/app/templates/src/app/components/Suspense/index.js (hunk 6 above)
Comment (damfinkel) [reply generators/app/templates/src/app/components/Suspense/index.js:4]: This component already existed. If we want to change it let's make it in another PR

**Review by damfinkel — APPROVED** [review]

**Review by wfolini — APPROVED** [review]
